spec/asyncapi.md

```diff
@@ -846,7 +846,7 @@ Field Name | Type | Description
 <a name="operationObjectExternalDocs"></a>externalDocs | [External Documentation Object](#externalDocumentationObject) \| [Reference Object](#referenceObject) | Additional external documentation for this operation.
 <a name="operationObjectBindings"></a>bindings | [Operation Bindings Object](#operationBindingsObject) \| [Reference Object](#referenceObject) | A map where the keys describe the name of the protocol and the values describe protocol-specific definitions for the operation.
 <a name="operationObjectTraits"></a>traits | [[Operation Trait Object](#operationTraitObject) &#124; [Reference Object](#referenceObject) ] | A list of traits to apply to the operation object. Traits MUST be merged using [traits merge mechanism](#traits-merge-mechanism). The resulting object MUST be a valid [Operation Object](#operationObject).
-<a name="operationObjectMessages"></a>messages | [[Reference Object](#referenceObject)] | A list of `$ref` pointers pointing to the supported [Message Objects](#messageObject) that can be processed by this operation. It MUST contain a subset of the messages defined in the [channel referenced in this operation](#operationObjectChannel), and MUST NOT point to a subset of message definitions located in the [Messages Object](#componentsMessages) in the [Components Object](#componentsObject) or anywhere else. **Every message processed by this operation MUST be valid against one, and only one, of the [message objects](#messageObject) referenced in this list.** Please note the `messages` property value MUST be a list of [Reference Objects](#referenceObject) and, therefore, MUST NOT contain [Message Objects](#messageObject). However, it is RECOMMENDED that parsers (or other software) dereference this property for a better development experience.
+<a name="operationObjectMessages"></a>messages | [[Reference Object](#referenceObject)] | A list of `$ref` pointers pointing to the supported [Message Objects](#messageObject) that can be processed by this operation. It MUST contain a subset of the messages defined in the [channel referenced in this operation](#operationObjectChannel), and MUST NOT point to a subset of message definitions located in the [Messages Object](#componentsMessages) in the [Components Object](#componentsObject) or anywhere else. **Every message processed by this operation MUST be valid against one, and only one, of the [message objects](#messageObject) referenced in this list.** Please note the `messages` property value MUST be a list of [Reference Objects](#referenceObject) and, therefore, MUST NOT contain [Message Objects](#messageObject). However, it is RECOMMENDED that parsers (or other software) dereference this property for a better development experience. <p>**Note**: excluding this property from the Operation implies that all messages from the channel will be included. Explicitly set the `messages` property to `[]` if this operation should contain no messages.</p>
```

Comment:
Out of curiosity, when is it useful to have an operation without messages?

Reply:
yeah I think we need different wording

@fmvilas this PR got opened because of slack convo https://asyncapi.slack.com/archives/C0230UAM6R3/p1747347552431919. I don't think @ctataryn means operation should contain no message.

@fmvilas you remember when we talked about v3 the `messages` in `operation` and `operation reply` - we always called them a filtering features. That you can have one channel that specifies all messages, but then on each operation level you specify what messages it relates too. And we also talked about situation that if in the operation you have no messages specified explicitly, it means this operation relates to all messages specified in the channel

Reply:
Thanks for the context, that helps 🙏 

I'm still curious to understand why `[]` should mean "this operation has no messages". Because we could easily say that `[]` is the equivalent of not setting the property at all, i.e., it means all the messages.

Reply:
Just out of pure coincidence I stumpled upon this issue. From my point of view I use it for HTTP requests without body such as GET, I find the easiest to define with empty array. 
![image](https://github.com/user-attachments/assets/972e4816-5f7f-4187-a3aa-c0642e2e66fc)

Have not figured out if there is an easier way 😄

Reply:
That's a great example, @jonaslagoni. Thanks!